### PR TITLE
Move colors definitions from the components less files to default.less

### PR DESCRIPTION
**Pull request by Nouzbe**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
^ I did not manage to get the Linter to work :/
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

----

**Related issue**: #11519 

**Goal**: Move all colors definitions from the components less indexes into `default.less` so that the whole style can be controlled with less variables. Specifically, it makes it easier (and less hacky) to create dark themed applications using Ant Design.

**Note**: New variables were introduced. This is just a proposal and the way I factorized colors and named variables can be discussed.

**Possible next step**: Adding a custom Lint rule to check that colors are not added to components less indexes in the future would make this more robust.

### CODE REVIEW

File: components/alert/style/index.less [hunk 1]
```diff
@@ -74,7 +74,7 @@
       color: @alert-close-color;
       transition: color .3s;
       &:hover {
-        color: #404040;
+        color: @icon-color;
```
Comment (afc163) [inline components/alert/style/index.less]: @icon-color-hover;

File: components/spin/style/index.less [hunk 2]
```diff
@@ -40,7 +40,7 @@
         top: 50%;
         width: 100%;
         padding-top: (@spin-dot-size - @font-size-base) / 2 + 2px;
-        text-shadow: 0 1px 2px #fff;
+        text-shadow: 0 1px 2px @alternate-shadow-color;
```
Comment (afc163) [inline components/spin/style/index.less]: `@shadow-color-inverse`, save rule as `@text-color-inverse`

File: components/style/themes/default.less [hunk 3]
```diff
@@ -12,6 +12,8 @@
 @error-color            : @red-6;
 @highlight-color        : @red-6;
 @warning-color          : @gold-6;
+@default-color          : #fff;
+@alternate-color        : #000;
```
Comment (afc163) [inline components/style/themes/default.less]: I don't really understand what `alternate` stands for which is sometimes white and sometimes black.

Comment (Nouzbe) [reply components/style/themes/default.less]: Maybe `default-light-color` and `default-dark-color` would be better, or even `@black` and `@white`.

I added these two in order to be able to replace the harcoded shades of grey by `shade(@black, x%)`, which has the benefit of working with any underlying background (as opposed to `#444` for instance, which assumes that the background is white and thus does not work if `@component-background` is assigned another value).

Comment (afc163) [reply components/style/themes/default.less]: `@black` & `@white` is nice and simple enough.

File: components/style/themes/default.less [hunk 4]
```diff
@@ -93,6 +97,7 @@
 // Border color
 @border-color-base      : hsv(0, 0, 85%);  // base border outline a component
 @border-color-split     : hsv(0, 0, 91%);  // split border inside a component
+@alternate-border-color : @white;
```
Comment (afc163) [inline components/style/themes/default.less]: @border-color-inverse

File: components/style/themes/default.less [hunk 5]
```diff
@@ -110,7 +115,8 @@
 @disabled-color-dark    : fade(#fff, 35%);
 
 // Shadow
-@shadow-color           : rgba(0, 0, 0, .15);
+@shadow-color           : fade(@black, 15%);
+@shadow-color-inverse : @component-background;
```
Comment (afc163) [inline components/style/themes/default.less]: It is better to align colon here.

File: components/input-number/style/index.less [hunk 6]
```diff
@@ -26,7 +26,7 @@
     width: 100%;
     font-weight: bold;
     &:active {
-      background: #f4f4f4;
+      background: @input-active-bg;
```
Comment (afc163) [inline components/input-number/style/index.less]: `@input-number-handler-active-bg`

File: components/style/themes/default.less [hunk 7]
```diff
@@ -386,7 +394,8 @@
 @card-padding-base: 24px;
 @card-padding-wider: 32px;
 @card-actions-background: @background-color-light;
-@card-shadow: 0 2px 8px rgba(0, 0, 0, .09);
+@card-shadow: 0 2px 8px fade(@black, 9%);
+@card-background: #cfd8dc;
```
Comment (afc163) [inline components/style/themes/default.less]: I can't find `@card-background`'s usage.

Comment (Nouzbe) [reply components/style/themes/default.less]: It is used in [components/card/style/index.less#L8](https://github.com/Nouzbe/ant-design/blob/feature-3.8.0/components/card/style/index.less#L8).

**Comment by Nouzbe** [comment]
@afc163 does it make it easier to you if I rebase with `feature-3.9.0`?

**Comment by Nouzbe** [comment]
Let's start fresh in #11779 (squashed changes and rebased `feature-3.9.0`)

**Pull request closed without merging**